ballerina/tests/12-custom-type-varray-test.bal

```diff
@@ -113,30 +113,247 @@ isolated function insertVarrayWithEmptyArray() returns sql:Error? {
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BYTEARR) values(${byteArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:DatabaseError {
+       test:assertTrue(result.message().includes("Error while executing SQL query: insert into " +
+       "TestVarrayTable(COL_BYTEARR) values( ? ). ORA-00600: internal error code"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_INTARR) values(${intArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :int[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BOOLARR) values(${boolArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :boolean[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_FLOATARR) values(${floatArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :float[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_DECIMALARR) values(${decimalArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :decimal[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+}
+
+type ArrayRecordType2 record {
+    int pk;
+    string?[] col_chararr;
+    byte[]?[] col_bytearr;
+    int?[] col_intarr;
+    boolean?[] col_boolarr;
+    float?[] col_floatarr;
+    decimal?[] col_decimalarr;
+};
+
 @test:Config {
     groups:["custom-varray"],
-    dependsOn: [insertVarrayWithEmptyArray]
+    dependsOn: [insertVarrayWithNullElements]
+}
+isolated function selectVarrayHavingNullElementsWithRecordType() returns error? {
+    ArrayRecordType2 arrayRecordInstance = {
+        pk : 4,
+        col_chararr : [null, null],
+        col_bytearr : [null, null],
+        col_intarr : [null, null],
+        col_boolarr : [null, null],
+        col_floatarr : [null, null],
+        col_decimalarr : [null, null]
+    };
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<ArrayRecordType2, error?> streamData = oracledbClient->query(
+        `SELECT pk, COL_CHARARR, COL_BYTEARR, COL_INTARR, COL_BOOLARR, COL_FLOATARR, COL_DECIMALARR
+         FROM TestVarrayTable WHERE pk = 4`);
+    record {|ArrayRecordType2 value;|}? data = check streamData.next();
```

Comment:
why we need to record here. Can't we just do `ArrayRecordType2 data = check streamData.next(); `

Reply:
We need record here. Can not use `ArrayRecordType2 data = check streamData.next();` code here. If so need to check in sql module level. also This proposed change would be a breaking change.